components/director/internal/domain/bundleinstanceauth/resolver.go

```diff
@@ -268,3 +272,96 @@ func (r *Resolver) RequestBundleInstanceAuthDeletion(ctx context.Context, authID
 
 	return r.conv.ToGraphQL(instanceAuth)
 }
+
+// CreateBundleInstanceAuth creates a BundleInstanceAuth for a Bundle with ID - bundleID from a given BundleInstanceAuthCreateInput
+func (r *Resolver) CreateBundleInstanceAuth(ctx context.Context, bundleID string, in graphql.BundleInstanceAuthCreateInput) (*graphql.BundleInstanceAuth, error) {
+	tx, err := r.transact.Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	defer r.transact.RollbackUnlessCommitted(ctx, tx)
+	ctx = persistence.SaveToContext(ctx, tx)
+
+	log.C(ctx).Infof("Creating BundleInstanceAuth for Bundle with id %q", bundleID)
+
+	bndl, err := r.bndlSvc.Get(ctx, bundleID)
+	if err != nil {
+		return nil, err
+	}
+
+	convertedIn, err := r.conv.CreateInputFromGraphQL(in)
+	if err != nil {
+		return nil, err
+	}
+
+	instanceAuthID, err := r.svc.CreateBundleInstanceAuth(ctx, bundleID, convertedIn, bndl.InstanceAuthRequestInputSchema)
+	if err != nil {
+		return nil, err
+	}
+
+	instanceAuth, err := r.svc.Get(ctx, instanceAuthID)
+	if err != nil {
+		return nil, err
+	}
+	log.C(ctx).Infof("Successfully created BundleInstanceAuth with id %q for Bundle with id %q", instanceAuthID, bundleID)
+
+	if err = tx.Commit(); err != nil {
+		return nil, err
+	}
+
+	return r.conv.ToGraphQL(instanceAuth)
+}
+
+// UpdateBundleInstanceAuth updates a BundleInstanceAuth with id for a Bundle with ID - bundleID from a given BundleInstanceAuthUpdateInput
+func (r *Resolver) UpdateBundleInstanceAuth(ctx context.Context, id string, bundleID string, in graphql.BundleInstanceAuthUpdateInput) (*graphql.BundleInstanceAuth, error) {
+	tx, err := r.transact.Begin()
+	if err != nil {
+		return nil, err
+	}
+
+	defer r.transact.RollbackUnlessCommitted(ctx, tx)
+	ctx = persistence.SaveToContext(ctx, tx)
+
+	log.C(ctx).Infof("Creating BundleInstanceAuth for Bundle with id %q", bundleID)
+
+	bndl, err := r.bndlSvc.Get(ctx, bundleID)
+	if err != nil {
+		return nil, err
+	}
+
+	convertedIn, err := r.conv.UpdateInputFromGraphQL(in)
+	if err != nil {
+		return nil, err
+	}
+
+	bndlInstAuth, err := r.svc.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	bndlInstAuth.SetFromUpdateInput(convertedIn)
+
+	if in.InputParams != nil {
+		log.C(ctx).Debugf("Validating BundleInstanceAuth request input for Bundle with id %q", bundleID)
+		if err = validateInputParamsAgainstSchema(convertedIn.InputParams, bndl.InstanceAuthRequestInputSchema); err != nil {
```

Comment:
Can we move the validation to the service layer as for Create so that the processing will be consistent

Reply:
It was discussed offline. At first, the validation was in the service layer and the `Update` method had `requestInputSchema` as a parameter which was the Bundle.InstanceAuthRequestInputSchema. The problem is that this Update is used in the runtime resolver to update instance auths, but there is no bundle and we can't provide the requestInputSchema for the new Update method. That's why I left the validation in the resolver.